packages/common/http/src/transfer_cache.ts

```diff
@@ -34,12 +38,34 @@ const ALLOWED_METHODS = ['GET', 'HEAD'];
 
 export function transferCacheInterceptorFn(
     req: HttpRequest<unknown>, next: HttpHandlerFn): Observable<HttpEvent<unknown>> {
-  const {isCacheActive} = inject(CACHE_STATE);
+  const {isCacheActive, ...globalTransferCacheOptions} = inject(CACHE_OPTIONS);
+
+  const uncachedMethod =
+      ((req.method === 'POST' && globalTransferCacheOptions.includePostRequests !== true) ||
+       (req.method !== 'POST' && !ALLOWED_METHODS.includes(req.method)));
+
+  // request-scoped options override the global ones
+  let transferCacheOptions = req.transferCache ?? globalTransferCacheOptions;
+
+  // converting the boolean value to the corresponding option
+  switch (transferCacheOptions) {
+    case true:
+      transferCacheOptions = {};
+      break;
+    case false:
+      transferCacheOptions = {filter: () => false};
+      break;
+  }
```

Comment:
I'd propose refactoring the code, so that we just handle parameters that we need (includeHeaders):
```suggestion
  // Review note: this is for minification purposes.
  const requestOptions = req.transferCache;
  
  // Review note: we can make this "if" a part of other conditions
  if (requestOptions === false || !globalOptions.filter?.(req)) {
    return next(req);
  }
  
  // ... later in the code when we know that caching should happen ...
  let includeHeaders = globalOptions.includeHeaders;
  if (typeof requestOptions === 'object' && typeof requestOptions.includeHeaders === 'boolean') {
    // Request-specific config takes precedence over the global config.
    includeHeaders = requestOptions.includeHeaders;
  }
  
  // ... use `includeHeaders` later ...
```